### ISSUE TITLE
UX: VPN switch jumps due to WiFi list height change

### ISSUE THREAD

**Issue by bbuhler** [elementary/wingpanel-indicator-network#134]
It happend to me already a few times, I accidentally clicked on a WiFi connection instead of the VPN switch, like actually intended, just because the WiFi list height changes and the VPN switch jumps down in the view.

![Screenshot from 2019-09-20 10 00 09](https://user-images.githubusercontent.com/2802675/65310290-4ae6f600-db8e-11e9-9874-8a313bf953d2.png)

Maybe we can improve the user experience by not changing the list height (respectively it's scroll container) as long as the mouse pointer is over the popover.

**Comment by janxkoci** [comment]
Happens to me constatntly. I think the best way would be to set a size and make the list scrollable (it already is scrollable, just fix the size).

**Comment by danrabbit** [comment]
We probably should either list VPNs first or have a tabbed or submenu interface of some kind

**Comment by itoshkov** [comment]
> We probably should either list VPNs first or have a tabbed or submenu interface of some kind

That would be an improvement for this specific case, but the same problem happens just with the Wi-Fi networks: Suppose you want to connect to the second one in the list. But just as you're clicking, the list changes and now you're clicking on another network, not the one you wanted.

One option to fix this would be to detect that the mouse pointer is in the window, and indicate that there are updates to the list, but without rearranging it. But that won't work for touch-driven interface.

Another option would be to disable the menu until it is updated. If the update happens in the background, even if the indicator is not shown, and the results are cached, it could work seamlessly.

**Comment by janxkoci** [comment]
> even if the indicator is not shown, and the results are cached, it could work seamlessly

I actually suspect this issue happens _because_ the wifi list is cached and then updates exactly when we want to click something.

**Comment by itoshkov** [comment]
> I actually suspect this issue happens _because_ the wifi list is cached and then updates exactly when we want to click something.

The whole point of caching would be not to have to wait for the list to get updated when you click on the icon, so I doubt that it causes the current problem.

**Comment by janxkoci** [comment]
So where does the initial list come from then? :thinking:

**Comment by itoshkov** [comment]
> So where does the initial list come from then?

I'm on elementary OS 6 now. The initial list of wi-fi connections consists of the one I'm on right now. When I click the icon, the list starts to populate. If I close it and open it soon after it remembers the populated list (cache) and in this case the UI works fine. 

The problem emerges, if I want to click on a Wi-Fi or VPN connection while the list is shown and is still populating. That's why I suggested to update the cache more often and to keep the list disabled, in the sense of non-clickable, if the cache isn't up-to-date yet.

**Comment by janxkoci** [comment]
> I'm on elementary OS 6 now. The initial list of wi-fi connections consists of the one I'm on right now.

I see this behaviour also on Hera (i.e. only the current wifi is shown initially), however I'm now in a house with very few wifi networks. I also remember that when I first upvoted this issue, I used to see more networks, like 10, and then the list expanded to a full list, bumping the VPN toggle from it's place.

> keep the list disabled, in the sense of non-clickable, if the cache isn't up-to-date yet

This sounds like a good idea to me. Unless there is a problem with updating and the list will stay non-clickable for too long.. That could be annoying.
